frontend/src/views/AdminPages/BranchesPage/BranchEditComponent/BranchEditComponent.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { saveBranch } from '../../../../actions/branchesActions';
+// @material-ui/core components
+import { makeStyles } from "@material-ui/core/styles";
+// core components
+import GridContainer from "components/Grid/GridContainer.js";
+import GridItem from "components/Grid/GridItem.js";
+import Card from "components/Card/Card.js";
+import CardBody from "components/Card/CardBody.js";
+import CardHeader from "components/Card/CardHeader.js";
+import CardFooter from "components/Card/CardFooter.js";
+import CustomInput from "components/CustomInput/CustomInput";
+import Button from "components/CustomButtons/Button.js";
+// assets
+import styles from "assets/jss/material-kit-react/views/loginPage.js";
+import Danger from "components/Typography/Danger";
+
+const useStyles = makeStyles(styles);
+
+export default function BranchEditComponent(props) {
+    const classes = useStyles();
+    const { branch, loadingSave, errorSave, onCloseClick, ...rest } = props;
+
+    // const {branch} = props; 
+    const [id, setId] = useState();
+    const [city, setCity] = useState();
+    const [description, setDescription] = useState();
+    const [image, setImage] = useState();
+    const [longitude, setLongitude] = useState();
+    const [latitude, setLatitude] = useState();
+    const dispatch = useDispatch();
```

Comment:
You might want put them into one state, to reduce the amount of repeating code `const [branch, setBranch] = useState(branch);`